lib/nostrum/struct/thread_member.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Nostrum.Struct.ThreadMember do
+  @moduledoc """
+  Struct representing a thread member object
+  """
+
+  alias Nostrum.Struct.{Guild, User}
+  alias Nostrum.{Snowflake, Util}
+
+  defstruct [
+    :id,
+    :user_id,
+    :join_timestamp,
+    :flags,
+    :guild_id
+  ]
+
+  @typedoc """
+  The id of the thread, omitted within `GUILD_CREATE` events
+  """
+  @typedoc since: "0.5.1"
+  @type id :: Snowflake.t() | nil
+
+  @typedoc """
+  The id of the user, omitted within `GUILD_CREATE` events
+  """
+  @typedoc since: "0.5.1"
+  @type user_id :: User.id() | nil
+
+  @typedoc """
+  The timestamp of when the user last joined the thread
+  """
+  @typedoc since: "0.5.1"
+  @type join_timestamp :: DateTime.t()
+
+  @typedoc """
+  Any user-thread settings
+  """
```

Comment:
This also needs the `@typedoc since: "0.5.1"`.

Alternatively, would it be more readable in the docs to simply have a `@moduledoc since: "0.5.1"` and keep `@typedoc since:` for any new struct fields that were added after the module was added?